### ISSUE TITLE
Discuss best way to manage tile-type extensibility.

### ISSUE THREAD

**Issue by samperumal** [samperumal/samperumal.github.io#2]

**Comment by mjrmua** [comment]
What's a tile-type?

**Comment by samperumal** [comment]
Good point. **Cell.*tt*** is meant to be that, but there's also **Item.*tt*** too.

Basically I might want to add 
 * a new tile-type (water, floor, lava)
 * a new item-type
 * a new monster-type

**Comment by mjrmua** [comment]
Delete tile-type as a concept and have items for water, wall, lava

**Comment by samperumal** [comment]
No, that complicates things like having something floating on water or lava. It's basically the difference between terrain, and what's on the terrain.

**Comment by mjrmua** [comment]
We have a different definition of "complicate"

**Comment by samperumal** [comment]
Ok, how so? I'm concerned about making things so generic that implementing new stuff actually becomes more complicated.

**Comment by mjrmua** [comment]
each an object with properties like as to how it interacts with systems
canBeAttacked, canWalkInto, canStandOn, , blocksVision etc 

so if you want a ring of phasing that lets you walk through things, you can use the same logic for walking through walls, goblins and treasure chests.
